proposals/20200504-falco-artifacts-scope.md

```diff
@@ -0,0 +1,191 @@
+
+As a project we will build, change, rename, and move files, documents, scripts, configurations according to this new state of the world.
+
+Inspired from [#1114](https://github.com/falcosecurity/falco/issues/1114) and many of the weekly community calls.
+
+## Terms
+
+**falco** 
+
+*Standalone Falco binary*
+
+**driver**
+
+*System call provider from the Linux kernel. Either (`bpf`, `module`, or `ptrace`)*
+
+**falco-driver-loader-script**
+
+
+*The bash script found [here](https://github.com/falcosecurity/falco/blob/master/scripts/falco-driver-loader) that tries to compile else download the kernel module.*
+
+**package**
+
+*An installable artifact that is operating system specific. All packages MUST be hosted on bintray for X86_64.*
+
+**image**
+
+*OCI compliant container image hosted on dockerhub with tags for every release.*
+ 
+
+# Packages
+
+Official packages for X86_64 only. All package names should contain a version which is not specified below.
+
+The following convention MUST be used for all packages.
+
+_If a package installs Falco as a daemon (systemd, init) it MUST contain `daemon`._
+
+_If a package installs the Falco kernel module it MUST contain `module`._
+
+_If a package installs the Falco BPF probe it MUST contain `bpf`._
+
+_If a package installs the Falco kernel module it MUST contain `module`._
+
+---
+
+### .deb
+
+ Falco running in debian like systems that will default to the kernel module.
+
+- falco-x86.deb
+     - `falco` only (default depends on `falco-module`)
+ - falco-daemon-x86.deb
+     - `falco` and script to configure as daemon (default depends on `falco-module`)
+ - falco-module-x86.deb
+     - `module` only
+ - falco-bpf-x86.deb
+     - `bpf` only
+ - falco-ptrace-x86.deb
+     - `ptrace` only
+
+### .rpm
+
+ Falco running in rpm like systems that will default to the kernel module.
+
+- falco-x86.deb
+     - `falco` only (default depends on `falco-module`)
+ - falco-daemon-x86.deb
+     - `falco` and script to configure as daemon (default depends on `falco-module`)
+ - falco-module-x86.deb
+     - `module` only
+ - falco-bpf-x86.deb
+     - `bpf` only
+ - falco-ptrace-x86.deb
+     - `ptrace` only
+
+
+### .tar.gz
+
+ - falco-src-x86.tar.gz
+     - No binaries
+     - `INSTALL` file
+ - falco-driver-src-x86.tar.gz
+     - `falco` and `/driver` with `Makefile`
+     - `INSTALL` file
+ - falco-driver-download-x86.tar.gz
+     - `falco` and `falco-driver-loader` download only
+ - falco-driver-full-x86.tar.gz
+     - `falco` and `falco-driver-loader` download else compile
+
+
+# Images
+
+The following convention MUST be used for all container images.
+
+_If a container image requires `--privileged` the name MUST have the `priviliged-` prefix._
+
+_If a container image is for development purproses it MUST have the `dev-` prefix._ 
+
+_If a container image is used to install something, then exit it MUST have the `-install` suffix.*_
+
+---
+
+ - falcosecurity/falco:TAG
+     - Runs `falco` userspace only
```

Comment:
I think that the current `falcosecurity/falco` image (which needs to run with `--privileged` to install the driver on the host) needs to stay as is and it needs to be called this way because it covers the use-case "run in 5 minutes".

When I want to try or demo a tool the first thing I do is try to use the docker image named after it.
I'd go away if I have 5 minutes and I have to go look for other docker images named in some other way to first run.

This was the reason (brought up by Loris during a community call) that we decided to keep it as is, and I think we all agreed on that.

Also in this case, I agree with you we need a better explanation in form of documentation.

Reply:
We should clarify that this image is NOT the production path. It's just there to cover the fast bootstrap/demo use-case.

Reply:
I agree that's mostly a documentation issue, not a naming problem.